snap/snapcraft.yaml

```diff
@@ -0,0 +1,112 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+---
+name: spack
+title: Spack
+icon: data/spack-logo.svg
+version: 0.20.0
+adopt-info: spack
+summary: A flexible package manager that supports multiple configurations.
+description: |
+  Spack is a multi-platform package manager that builds and installs multiple
+   versions and configurations of software. It works on Linux, macOS, and many
+   supercomputers. Spack is non-destructive: installing a new version of a
+   package does not break existing installations, so many configurations of the
+   same package can coexist.
+
+  Spack offers a simple "spec" syntax that allows users to specify versions and
+   configuration options. Package files are written in pure Python, and specs
+   allow package authors to write a single script for many different builds of
+   the same package. With Spack, you can build your software all the ways you
+   want to.
+license: Apache-2.0
+website: https://spack.io/
+base: core22
+grade: stable
```

Comment:
I think its worth not quite marking the grade as stable at this point. It will prevent a publish to the candidate/stable channels - but I think this is exactly what we want here.